server/src/main/java/com/genymobile/scrcpy/ScreenEncoder.java

```diff
@@ -148,15 +167,20 @@ private boolean encode(MediaCodec codec, FileDescriptor fd) throws IOException {
         boolean eof = false;
         MediaCodec.BufferInfo bufferInfo = new MediaCodec.BufferInfo();
 
-        while (!consumeRotationChange() && !eof) {
-            int outputBufferId = codec.dequeueOutputBuffer(bufferInfo, -1);
+        while (!consumeRotationChange() && !consumeKeyFrameRequest() && !eof) {
+            // NOTE(frankleonrose): Timeout after 10ms to respond to key frame request.
+            int outputBufferId = codec.dequeueOutputBuffer(bufferInfo, 10000);
             eof = (bufferInfo.flags & MediaCodec.BUFFER_FLAG_END_OF_STREAM) != 0;
-            try {
-                if (consumeRotationChange()) {
```

Comment:
Does removing this handle cause any concerns? We're not breaking now if there was a rotation change, so perhaps that has a possibility of affecting keyframe captures?

I guess even if that is true the odds of it being a problem in our use case is negligible.

Reply:
The surrounding `while` includes the `consumeRotationChange()` check. It was necessary to check within the loop in the earlier version, because the code used to block within `codec.dequeueOutputBuffer` and a rotation might have happened since then. We pop out of `codec.dequeueOutputBuffer` with some frequency, so the `while` is sufficient.